### DISCUSSION TITLE
Support multiple database for Appwrite

### DISCUSSION THREAD

**Discussion by nguyenthanhtien** [appwrite/appwrite#7849] [Q&A]
I want ask can support another database for Appwrite, can support another database type?
MariaDB imposes a row-size limit of 65,535 bytes for the combined sizes of all columns. If the table contains [BLOB](https://mariadb.com/kb/en/blob/) or [TEXT](https://mariadb.com/kb/en/text/) columns, these only count for 9 - 12 bytes in this calculation, given that their content is stored separately.

**Comment by wellkilo** [comment]
For self-hosted Appwrite, the current installer/runtime supports three database adapters: **PostgreSQL, MariaDB, and MongoDB**. Select one with `_APP_DB_ADAPTER` during installation/deployment; it is not an arbitrary database-driver plug-in point.

Example environment setting:

```env
_APP_DB_ADAPTER=postgresql
```

and configure the corresponding `_APP_DB_HOST`, `_APP_DB_PORT`, `_APP_DB_USER`, `_APP_DB_PASS`, and `_APP_DB_SCHEMA` values for that database service. The current installer validates the adapter against exactly `postgresql`, `mariadb`, or `mongodb` ([runtime validation](https://github.com/appwrite/appwrite/blob/294eb16a9639487f57537be3fa6e78f78f2b955e/src/Appwrite/Platform/Installer/Runtime/State.php#L200-L210)); the environment definition exposes the same choices ([config](https://github.com/appwrite/appwrite/blob/294eb16a9639487f57537be3fa6e78f78f2b955e/app/config/variables.php#L496-L510)).

Important boundaries:

- This is a self-hosted deployment choice, not something you switch for Appwrite Cloud.
- Do not point a running installation at a different empty database and expect data to migrate. Changing adapters requires an Appwrite-supported migration/export-import plan and backups.
- VectorsDB has a separate adapter and currently requires PostgreSQL; it is not controlled solely by the primary `_APP_DB_ADAPTER`.
- Supporting a fourth engine requires an Appwrite/Utopia database adapter plus capability and migration work; a compatible SQL dialect alone is not sufficient.

If PostgreSQL/MongoDB/MariaDB covers the database type you meant, please mark this as accepted; otherwise name the desired engine and the constraint it solves so compatibility can be evaluated concretely.
